### DISCUSSION TITLE
How to fetch data using SSR, and also use hooks in same component or parent component?

### DISCUSSION THREAD

**Discussion by CodeStarGM** [vercel/next.js#58165] [Help]
### Summary

I'm currently developing an application, and I've reached a point where I need to fetch data on the server through SSR (Server-Side Rendering). Additionally, I want to integrate hooks for data filtering, allowing users to filter the data as needed.

I've been searching for a solution, but I haven't found any helpful resources so far. I even attempted to create a parent component to include the SSR component and implement hooks. However, I encountered an error after a few seconds, which reads: 'Uncaught Error: async/await is not yet supported in Client Components, only Server Components. This error is often caused by accidentally adding 'use client' to a module that was originally written for the server.'

Any assistance or guidance on this matter would be greatly appreciated.

**Comment by mutasim77** [comment]
I think you can simply achieve this by creating a client-side component and passing the data to it, and then working with hooks

```js
'use client'
import { useEffect, useState, useRef } from 'react'; // hooks here

const MyComponent = ({ initialData }) => {
  const [data, setData] = useState(initialData); 
  const inputRef = useRef(null); 

  useEffect(() => {
    // You can add client-side logic and hooks here
  }, [data]);

  return (
    <div>
      {/* Render your data here */}
    </div>
  );
};

export default MyComponent;
```

```js
'use server' // anyways by default it's server
 
import MyComponent from '../components/MyComponent';

const fetchDataOnServer = async () => {
  const data = await fetch('your-api');
  return data.json();
};

export async function getServerSideProps() {
  const initialData = await fetchDataOnServer();

  return {
    props: { initialData },
  };
}

const MyPage = ({ initialData }) => {
  return <MyComponent initialData={initialData} />; // <- here
};

export default MyPage;
```

**Comment by kalossi** [comment]
Could you provide an example how you arrange your directories where those files are? For example app/pages.js and app/components/MyComponent.js does not work since you cant call server side components in .app/ root.

**Reply by DarkoTrpevski** [reply]
I think he's referring that MyComponent is coming from outside of the app directory. Like the file structure is:
app => page.jsx
src/components/MyComponent.jsx
